lib/visitor.js

```diff
@@ -443,15 +444,40 @@ class Visitor {
         }, '}')
     }
 
+    /**
+     * @param {string} fq
+     * @param {{ name: string; doc?: string; operations?: {returns: any }}} service
+     */
     #printService(fq, service) {
         LOG.debug(`Printing service ${fq}:\n${JSON.stringify(service, null, 2)}`)
+        // @ts-ignore
         const { namespace } = this.entityRepository.getByFq(fq)
         const file = this.fileRepository.getNamespaceFile(namespace)
-        // service.name is clean of namespace
-        file.services.buffer.add(`export default { name: '${service.name}' }`)
+        const buffer = file.services.buffer
+
+        this.#renderDoc(service.doc, buffer)
+        // file.addImport(new Path(['cds'], '')) TODO make sap/cds import work
+        buffer.addIndentedBlock('export default class extends cds.Service {', () => {
+            Object.entries(service.operations||{}).forEach(([name, {doc}]) => {
+                this.#renderDoc(doc, buffer)
+                buffer.add(`declare ${name}: typeof ${name}`)
+            })
+        }, '}')
         file.addService(service.name)
     }
 
+    /**
+     * @param {string|undefined} doc - the documentation to add
+     * @param {import('./file.js').Buffer} buffer - the buffer to add the documentation to
+     */
+    #renderDoc(doc, buffer = new Buffer()) {
+        if (!doc) return ''
+        buffer.add('/**')
+        doc.split(/\r?\n/).forEach(line => { buffer.add(` * ${line}`) })
+        buffer.add(' */')
+        return buffer.join()
+    }
```

Comment:
```suggestion
    #renderDoc(doc, buffer) {
        docify(doc).forEach(d => buffer.add(d))
    }
```
I guess?
Or just inline it at the appropriate questions